### DISCUSSION TITLE
Google Stadia Controller Support

### DISCUSSION THREAD

**Discussion by PhilRoll** [open-goal/jak-project#2964] [General]
Hi everyone!
i have been trying the statia controller in bluetooth mode and i have noticed that the controller is not recognized correctly. 
some buttons are not recognized at all despite trying to remap the controller.
here a link on the specifications of the controller in question: 
[https://www.pcgamingwiki.com/wiki/Controller:Stadia_Controller](url)

I hope support for this controller will be added in future versions!

**Comment by Kuitar5** [comment]
Have you tried something [like this (or similar)](https://github.com/walkco/stadia-vigem) to emulate xinput? (xbox controller)
Iirc the controller supports are not being added manually for this project but use another maintained library (SDL2) so you'd have more luck trying to get it added over on that project.

**Reply by PhilRoll** [reply]
on windows environment this works. but this workaround cannot be applied on the Batocera OS distro (linux distro on which I am testing the contoller)

**Comment by xTVaser** [comment]
As of 7 months ago atleast, it doesn't seem like the stadia controller was well supported by SDL2 and someone had to configure their own mapping to get the controller to work properly https://www.reddit.com/r/Stadia/comments/10muihx/linux_map_the_bluetooth_stadia_controller_to_sdl2/

We pull in the latest community-maintained list of mappings from here https://github.com/gabomdq/SDL_GameControllerDB so if fixing the mapping works for you, you should consider contributing it there so we'll eventually consume it.

You can find the mapping file in `<PATH_TO_DATA_FOLDER>/game/assets/sdl_controller_db.txt`

**Reply by PhilRoll** [reply]
I have tried, but the problem persists. some inputs are taken correctly, some are not, and some are not recognized at all ( i.e. A or R2 button).
Maybe on Batocera OS (when i start the game) the os use another SDL controller mapping/these configs conflict with opengoal SDL?
because I also tried temporarily deleting the "sdl_controller_db.txt" file, but the controller performs in the same way

**Comment by PhilRoll** [comment]
Hi guys, i solved the iusse. It was a BatoceraOS "problem". 
To fix it i added this command line on the sh script needed for run the game:
`unset SDL_GAMECONTROLLERCONFIG `

I hope this can help other people! Thanks a lot for the support!

**Comment by xTVaser** [comment]
ah cool, we should detect those env-vars and log a warning that controller support may be impacted, thanks for tracking that down.
